lib/reporting/reports/enterprise_fee_summary/enterprise_fees_with_tax_report_by_order.rb

```diff
@@ -0,0 +1,259 @@
+            distributor: :distributor,
+            order_cycle: :order_cycle,
+            order_number: :order_number,
+            enterprise_fee_name: :enterprise_fee_name,
+            enterprise_fee_type: :enterprise_fee_type,
+            enterprise_fee_owner: :enterprise_fee_owner,
+            tax_category: :tax_category,
+            tax_rate_name: :tax_rate_name,
+            tax_rate: :tax_rate_amount,
+            total_excl_tax: :total_excl_tax,
+            tax: :tax,
+            total_incl_tax: :total_incl_tax,
+            customer_first_name: :customer_first_name,
+            customer_last_name: :customer_last_name,
+            customer_code: :customer_code,
+            customer_email: :customer_email
+          }
+        end
+
+        def rules
+          [
+            { group_by: :distributor },
+            { group_by: :order_cycle },
+            {
+              group_by: :order_number,
+              summary_row: proc do |_key, items, _rows|
+                item = items.first
+                order = item.second
+                enterprise_fees = order.all_adjustments.enterprise_fee.sum(:amount)
+                {
+                  total_excl_tax: enterprise_fees - order.enterprise_fee_tax(included: true),
+                  tax: order.enterprise_fee_tax,
+                  total_incl_tax: enterprise_fees + order.enterprise_fee_tax(added: true),
+                  customer_first_name: order.customer&.first_name,
+                  customer_last_name: order.customer&.last_name,
+                  customer_code: order.customer&.code,
+                  customer_email: order.customer&.email
+                }
+              end
+            }
+          ]
+        end
+
+        def distributor(query_result_row)
+          order(query_result_row).distributor&.name
+        end
+
+        def order_cycle(query_result_row)
+          order(query_result_row).order_cycle&.name
+        end
+
+        def order_number(query_result_row)
+          order(query_result_row).number
+        end
+
+        def enterprise_fee_name(query_result_row)
+          enterprise_fee(query_result_row).name
+        end
+
+        def enterprise_fee_type(query_result_row)
+          enterprise_fee(query_result_row).fee_type
+        end
+
+        def enterprise_fee_owner(query_result_row)
+          enterprise_fee(query_result_row).enterprise.name
+        end
+
+        def tax_category(query_result_row)
+          tax_rate(query_result_row)&.tax_category&.name
+        end
+
+        def tax_rate_name(query_result_row)
+          tax_rate(query_result_row)&.name
+        end
+
+        def tax_rate_amount(query_result_row)
+          tax_rate(query_result_row)&.amount
+        end
+
+        def total_excl_tax(query_result_row)
+          order_id = order(query_result_row).id
+          enterpise_fee_id = enterprise_fee_id(query_result_row)
+          amount = Spree::Adjustment.enterprise_fee
+            .where(order_id: order_id)
```

Comment:
We could actually pass the `Order` object in directly, and Rails automatically matches the ID:

```suggestion
            .where(order: order(query_result_row))
```

Reply:
done